schema/deploy/warehouse/identifier/triggers/barcode-distance-check.sql

```diff
@@ -23,6 +24,11 @@ create or replace function warehouse.identifier_barcode_distance_check() returns
         raise debug 'Checking new barcode "%" is at least % substitutions away from existing barcodes (except old barcode "%")',
             NEW.barcode, minimum_distance, old_barcode;
 
+        -- Populate the row's array of barcode slices to compare with existing rows.
+        -- This greatly reduces the number of barcodes against which the
+        -- `hamming_distance_lte` function is applied.
+        NEW.slices = public.barcode_slices(NEW.barcode);
```

Comment:
Do we need to think about the value of `OLD.slices` here?

I guess the current behaviour would be that you can never set `slices` yourself, either on insert or update; it's always overwritten by this assignment. Is that right?

Reply:
Thanks for seeing this. In the current state someone could update slices with whatever value they want.  I'm now trying various approaches to prevent someone from updating slices in a way that leaves its value not matching barcode_slices(barcode). I really loved the idea of making this a function index instead of a column. (That's such a cool pg feature.) Then we'd have no column to worry about. But pg doesn't use that index for the intersection operation. It uses the index if I say an explicit `where barcode_slices(barcode) = '{blah'` but not `where barcode_slices(barcode) && barcode_slices(NEW.barcode)`. 
Preventing violating updates would be a similar problem if we stored the slices in a separate table; we'd have to make sure that someone couldn't update that table wrong. 
This is not resolved in any commit yet. I'm still tinkering for the cleanest approach.

Reply:
Thanks for the discussion this morning. Using a functional index instead of having a `slices` column in the table gets rid of the operational issues we'd have to deal with around people updating the slices column. And minting is lots faster with the index. I'll add that info as a comment into this PR. When it's time to merge I'll update the commit message with the new numbers.